### PR TITLE
Fix h5mesh data attribute conflict

### LINKED ISSUE #144: H5Mesh conflict between cuba data tables
It looks that the hdf5 backed mesh defines the two tables to hold the container data attirbute and the items data content on a table node with the same name. See lines:

https://github.com/simphony/simphony-common/blob/master/simphony/io/h5_mesh.py#L128
https://github.com/simphony/simphony-common/blob/master/simphony/io/h5_mesh.py#L791

The problem was not detected because the tests do not really check for that sort of interaction.

### PR DESCRIPTION
**Pull request by itziakos**
this PR fixes #144

### CODE REVIEW

File: simphony/io/tests/test_h5_mesh.py [hunk 1]
```diff
@@ -106,5 +106,53 @@ def container_factory(self, name):
...
+        # when
+        self.assertEqual(len(group._f_list_nodes()), 6)
+        self.assertIsInstance(group.cells, tables.Table)
+        self.assertIsInstance(group.points, tables.Table)
+        self.assertIsInstance(group.edges, tables.Table)
+        self.assertIsInstance(group.faces, tables.Table)
+        self.assertIsInstance(group.data, tables.Table)
+        self.assertIsInstance(group.item_data, tables.Table)
+
+    def test_mesh_layout_with_new_proxy(self):
```
Comment (roigcarlo) [inline simphony/io/tests/test_h5_mesh.py]: I am not sure what is the purpose of this test.
Is this planned to check the duplicated tables of https://github.com/simphony/simphony-common/issues/148?

File: simphony/io/tests/test_h5_mesh.py [hunk 2]
```diff
@@ -106,5 +106,53 @@ def container_factory(self, name):
...
+        # then
```
Comment (roigcarlo) [inline simphony/io/tests/test_h5_mesh.py]: Given

**Comment by roigcarlo** [comment]
Just a couple of comments, but as soon as it passes Travis :+1:

File: simphony/io/tests/test_h5_mesh.py (hunk 1 above)
Comment (itziakos) [inline simphony/io/tests/test_h5_mesh.py]: It is me (going and little overboard) checking that we do not create multiple nodes when we create multiple proxies of a 
already existing group. It is not directed for #148

**Comment by nathanfranklin** [comment]
Just one comment: the hdf5-psuedo-uml diagrams have to be updated with the new name ("item_data")

File: simphony/io/tests/test_h5_mesh.py (hunk 1 above)
Comment (roigcarlo) [inline simphony/io/tests/test_h5_mesh.py]: Ok nice, thx

**Comment by nathanfranklin** [comment]
:+1:

**Pull request merged**